### PR TITLE
implement null safe equal <=> operator

### PR DESCRIPTION
**Pull request by ginter**
NULL-safe equal. This operator performs an equality comparison like the [=]noperator, but returns 1 rather than NULL if both operands are NULL, and 0 rather than NULL if one operand is NULL.

The [<=>] operator is equivalent to the standard SQL IS NOT DISTINCT FROM operator.
https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to

### CODE REVIEW

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -423,6 +424,7 @@ class Tokenizer(metaclass=_Tokenizer):
         "<=": TokenType.LTE,
         "<>": TokenType.NEQ,
         "!=": TokenType.NEQ,
+        "<=>": TokenType.NULLSAFE_EQ,
```
Comment (tobymao) [inline sqlglot/tokens.py:427]: also add "IS NOT DISTINCT FROM" ?

File: tests/test_transpile.py [hunk 2]
```diff
@@ -350,3 +350,7 @@ def unsupported(level):
         with self.assertRaises(UnsupportedError) as ctx:
             unsupported(ErrorLevel.IMMEDIATE)
         self.assertEqual(str(ctx.exception).count(error), 1)
+
+    def test_nullsafe_eq(self):
```
Comment (tobymao) [inline tests/test_transpile.py:354]: let's not put this here, let's put it in dialects and then test both ways

File: sqlglot/tokens.py (hunk 1 above)
Comment (tobymao) [reply sqlglot/tokens.py:427]: there's also is distinct from what is that?

File: sqlglot/optimizer/simplify.py [hunk 3]
```diff
@@ -257,6 +257,12 @@ def _simplify_binary(expression, a, b):
                 return TRUE if not_ else FALSE
             if a == NULL:
                 return FALSE if not_ else TRUE
+    elif isinstance(expression, exp.NullSafeEQ):
+        if a == b:
```
Comment (tobymao) [inline sqlglot/optimizer/simplify.py:261]: is this correct? how do you know if column a == column b? you don't know the values

File: sqlglot/parser.py [hunk 4]
```diff
@@ -1632,12 +1633,16 @@ def _parse_range(self):
...
-        this = self.expression(
-            exp.Is,
-            this=this,
-            expression=self._parse_null() or self._parse_boolean(),
-        )
-        return self.expression(exp.Not, this=this) if negate else this
+        if self._match(TokenType.DISTINCT_FROM):
+            klass = exp.NullSafeEQ if negate else exp.NullSafeNEQ
+            return self.expression(klass, this=this, expression=self._parse_expression())
+        else:
```
Comment (tobymao) [inline sqlglot/parser.py:1639]: you can remove the else now

File: sqlglot/tokens.py [hunk 5]
```diff
@@ -145,6 +146,7 @@ class TokenType(AutoName):
     DESCRIBE = auto()
     DETERMINISTIC = auto()
     DISTINCT = auto()
+    DISTINCT_FROM = auto()
```
Comment (ginter) [inline sqlglot/tokens.py:149]: unsure if it's preferable to piggy back off of the `IS` or expand both operators fully here

Comment (tobymao) [reply sqlglot/tokens.py:149]: this is good

File: sqlglot/optimizer/simplify.py (hunk 3 above)
Comment (ginter) [reply sqlglot/optimizer/simplify.py:261]: I was following the same logic as the EQ case below it except NULL <=> NULL is TRUE so I put it before the null check
```
    if isinstance(expression, exp.EQ) and a == b:
        return TRUE
```

Comment (tobymao) [reply sqlglot/optimizer/simplify.py:261]: brain fart, this looks correct, because a == b means that the column is the same

**Review by tobymao — APPROVED** [review]

**Pull request merged**